### PR TITLE
[FIX] mail: show message action list next to attachment

### PR DESCRIPTION
**Pull request by alexkuhn**
Before this commit, when a message had no text, the action list was placed above the message content, including attachments.

This made it hard to find the action list for these messages, as all action lists of any messages are expected to always be next to the 1st line of the message content.

This commit fixes the issue by placing action list next to text content if there are any (same as before), but if there are no text, then the actions are displayed next to attachments.

The template requiring defining a shared container for text content and attachments, so that action list comes after. Also when the message has some text and some attachments, the action list is still expected to come inline of text content, which is why this commit conditionally places the action list accordingly.

Part of Task-3340653

Before
before
After
after

### CODE REVIEW

File: addons/mail/static/src/core_ui/message.js [hunk 1]
```diff
@@ -253,6 +253,15 @@ export class Message extends Component {
...
 
+    get hasSomeFilePreviewsVisible() {
+        return Boolean(
+            this.message.attachments.length > 0 ||
+                (this.message.linkPreviews.length > 0 &&
+                    this.store.hasLinkPreviewFeature &&
+                    !this.message.linkPreviewSquash)
+        );
+    }
+
```
Comment (phenix-factory) [inline addons/mail/static/src/core_ui/message.js:264]: This is not used?

Comment (alexkuhn) [reply addons/mail/static/src/core_ui/message.js:264]: Indeed, residuals of attempt to conditional based on shown attachments. I did the opposite instead with conditional on presence of text :)

File: addons/mail/static/src/core/message_model.js [hunk 2]
```diff
@@ -181,7 +181,7 @@ export class Message {
     }
 
     get hasTextContent() {
-        return (this.editDate && this.attachments.length) || !this.isBodyEmpty;
+        return /*(this.editDate && this.attachments.length) || */ !this.isBodyEmpty;
```
Comment (oomsveta) [inline addons/mail/static/src/core/message_model.js:184]: remove this comment

File: addons/mail/static/src/core_ui/message.xml [hunk 3]
```diff
@@ -87,82 +87,89 @@
...
-                                            <DropdownItem t-if="editable" class="'px-2 d-flex align-items-center rounded-0'" title="'Edit'" onSelected="() => this.onClickEdit()"><i class="fa fa-lg fa-fw fa-pencil pe-2"/>Edit</DropdownItem>
-                                            <DropdownItem t-if="deletable" class="'px-2 d-flex align-items-center rounded-0'" title="'Delete'" onSelected="() => this.onClickDelete()"><i class="fa fa-lg fa-fw fa-trash pe-2"/>Delete</DropdownItem>
-                                        </t>
-                                    </Dropdown>
+                                    </t>
+                                    <t t-if="message.hasTextContent and !env.inChatWindow" t-call="mail.Message.actions"/>
                                 </div>
+                                <AttachmentList
+                                    t-if="message.attachments.length > 0"
+                                    attachments="message.attachments.map((a) => a)"
```
Comment (oomsveta) [inline addons/mail/static/src/core_ui/message.xml:123]: looks no op to me

Comment (alexkuhn) [reply addons/mail/static/src/core_ui/message.xml:123]: We need this for the good working of `reactive`
cc @seb-odoo

File: addons/mail/static/src/core/message_model.js (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/core/message_model.js:184]: The feature of edit date was not fully removed in 16.3. We just hide the edited label.
This is the minimal diff to disable it for the time being and fix it once and for all in master.
The less diff the better, and sooner or later we will need to add it back, as the JS is fine but the python needs to change (i.e. not use `create_date` / `write_date`)

**Review by oomsveta — APPROVED** [review]
robodoo r+

File: addons/mail/static/src/core_ui/message.xml (hunk 3 above)
Comment (oomsveta) [reply addons/mail/static/src/core_ui/message.xml:123]: would have deserved a comment then

**Comment by alexkuhn** [comment]
@robodoo rebase-ff

**Pull request closed without merging**